puppetry/kinectv2/pygame_button.py

```diff
@@ -0,0 +1,183 @@
+        pygame.draw.rect(surface, self._hover_color, (self._left, self._top, self._width, self._height))
+        rendered_text = self._font.render(F"{self._label}",True, self._text_color)
+        shift = 4 if mouse_down else 0
+        surface.blit(rendered_text, [self._text_x + shift, self._text_y + shift])
+
+
+    def mouse_over_button(self, mouse_pos):
+        """ Return True if mouse is over the button space """
+        mob = mouse_pos[0] > self._left and mouse_pos[0] < self._right and \
+              mouse_pos[1] > self._top and mouse_pos[1] < self._bottom
+        # button_logger.msg(f'MOB self._label : '
+        #           f'({self._left},{self._top}) x ({self._right}, {self._bottom}), '
+        #           f'mouse {mouse_pos} over: {mob}'
+        return mob
+
+    def handle_mouse(self, surface, mouse_pos, mouse_down):
+        """ Process the given mouse state once per frame
+            deal with hovering and clicks """
+        mob = self.mouse_over_button(mouse_pos)
+        if mob:
+            if self._mouse_down and not mouse_down and \
+                self._click_callback is not None:
+                # mouse was down, now up -> click
+                self._click_callback(self._name)
+            self._mouse_down = mouse_down
+            self.draw_hover(surface, mouse_down)
+        else:       # Mouse outside
+            self._mouse_down = False
+            self.draw_normal(surface)
+
+
+#---------------------------------------------------------------
+
+class PYGButtonManager():
+    """ One Button manager to rule them all
+        To do:
+            delete_button(self, name)
+    """
+    def __init__(self, surface):
+        self._surface = surface
+        self._buttons = {}
+
+    def create_button(self, new_info, click_callback = None):
+        """ Create a new button"""
+
+        name = new_info['name']
+        if name not in self._buttons:
+            button_info = {'name' : name,
+                        'text_size' : BTN_TEXT_SIZE,
+                        'text_color' : BTN_TEXT_COLOR,
+                        'button_color' : BTN_COLOR,
+                        'hover_color' : BTN_HOVER_COLOR}
+
+            if 'text_size' in new_info:
+                button_info['text_size'] = new_info['text_size']
+            if 'text_color' in new_info:
+                button_info['text_color'] = new_info['text_color']
+            if 'button_color' in new_info:
+                button_info['button_color'] = new_info['button_color']
+            if 'hover_color' in new_info:
+                button_info['hover_color'] = new_info['hover_color']
+
+            button_info['label'] = new_info['label']
+            button_info['center_x'] = new_info['center_x']
+            button_info['center_y'] = new_info['center_y']
+            button_info['width'] = new_info['width']
+            button_info['height'] = new_info['height']
```

Comment:
Could the mapping in lines 154-167 be replaced with `dict.update()` or something like `{'text_size': BTN_TEXT_SIZE, **new_info}`?